packages/mobile/src/walletConnect/request.ts

```diff
@@ -29,14 +31,41 @@ export function* handleRequest({ method, params }: { method: string; params: any
   yield call(unlockAccount, account)
 
   switch (method) {
-    case SupportedActions.eth_signTransaction:
-      return (yield call(wallet.signTransaction.bind(wallet), params[0])) as EncodedTransaction
+    case SupportedActions.eth_signTransaction: {
+      // IMPORTANT: We need to normalize the transaction parameters
+      // WalletConnect v1 utils currently strips away important fields like `chainId`, `feeCurrency`, `gatewayFee` and `gatewayFeeRecipient`
+      // See https://github.com/WalletConnect/walletconnect-monorepo/blame/c6b26481c34848dbc9c49bb0d024bda907ec4599/packages/helpers/utils/src/ethereum.ts#L66-L86
+      // Also the dapp developer may have omitted some of the needed fields,
+      // so it's nice to be flexible and still allow the transaction to be signed (and sent) successfully
+
+      const kit: ContractKit = yield call(getContractKit)
+      const normalizer = new TxParamsNormalizer(kit.connection)
+      const rawTx = { ...params[0] }
+      // For now if `feeCurrency` is not set, we don't know whether it was stripped by WalletConnect v1 utils or intentionally left out
+      // to use CELO to pay for fees
+      if (!rawTx.feeCurrency) {
+        // This will use CELO to pay for fees if the user has a balance,
+        // otherwise it will fallback to the first currency with a balance
+        const feeCurrency: Currency = yield call(chooseFeeCurrency, Currency.Celo)
+        // Pass undefined to use CELO to pay for gas.
+        const feeCurrencyAddress: string | undefined =
+          feeCurrency === Currency.Celo ? undefined : yield call(getCurrencyAddress, feeCurrency)
+
+        rawTx.feeCurrency = feeCurrencyAddress
+        // Note: we're resetting gas and gasPrice here because if the feeCurrency has changed, we need to recalculate them
+        rawTx.gas = undefined
+        rawTx.gasPrice = undefined
+      }
+      const tx: CeloTx = yield call(normalizer.populate.bind(normalizer), rawTx)
```

Comment:
It looks like this eventually makes an `eth_gasPrice` RPC call to the supporting Geth node. This will suggest an appropriate gas price, which by default is 5x the GPM. So this seems like it should result in a setting that will work.

https://github.com/celo-org/celo-blockchain/blob/8cd59acbb3ff70033c45867d2d62dbda0032c36a/contracts/gasprice_minimum/gasprice_minimum.go#L55-L60

Reply:
Great! Thanks for checking 💪